### PR TITLE
Docs: Improve pandas guide

### PR DESCRIPTION
**Pull request by NickCrews**
Couple small tweaks to reorganize and add some missing bits to
the pandas guide.

### CODE REVIEW

File: docs/ibis-for-pandas-users.ipynb [hunk 1]
```diff
@@ -31,6 +31,7 @@
    ]
```
Comment (NickCrews) [inline docs/ibis-for-pandas-users.ipynb:31]: <p>It's not obvious in the reviewnb UI, but I changed the header level on some of these headings</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis/pull/6447/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='87'/>

Comment (cpcloud) [inline docs/ibis-for-pandas-users.ipynb:31]: <p>There's nothing _in principle_ that prevents t.my_column.head(1) from working. It's the combination of that expression with almost anything else that doesn't make sense. The technical challenge is that we don't have a straightforward place to put a rewrite rule like</p><p> </p><pre class="ql-syntax" spellcheck="false">t[column].head(N) =&gt; t.head(N)[column]
</pre><p> </p><p>Is there some language we can add here to indicate that this particular operation isn't a fundamental limitation of ibis?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis/pull/6447/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (cpcloud) [inline docs/ibis-for-pandas-users.ipynb:31]: <p>Should this pound sign be visible here in the rendered output?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis/pull/6447/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='99'/>

**Review by cpcloud — APPROVED** [review]

File: docs/ibis-for-pandas-users.ipynb (hunk 1 above)
Comment (cpcloud) [reply docs/ibis-for-pandas-users.ipynb:31]: I'll fix this up!

Comment (cpcloud) [reply docs/ibis-for-pandas-users.ipynb:31]: This is fine for now!

**Comment by NickCrews** [comment]
Oops sorry this fell off my radar. Indeed I think both your suggestions are good.

**Pull request merged**